sklearn/mixture/dpgmm.py

```diff
@@ -566,18 +566,31 @@ def fit(self, X, y=None):
                                             np.trace(self.scale_[k]))
                 self.bound_prec_ *= 0.5
 
-        logprob = []
+        # EM algorithms
+        current_log_likelihood = None
         # reset self.converged_ to False
         self.converged_ = False
+
+        # this line should be removed when 'thresh' is removed in v0.18
+        tol = (self.tol if self.thresh is None
+               else self.thresh / float(n_samples))
```

Comment:
Instead of using the  `/ float(stufff)`  idiom, it's possible to do `from __future__ import division` at the top of th file.